### ISSUE TITLE
Result if far too different

### ISSUE THREAD

**Issue by launchpad-issue-importer** [scour-project/scour#79]
Optimizing loses some elements.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2011-05-19T16:04:38Z
 * owner: andresm-plast
 * the launchpad url was https://bugs.launchpad.net/bugs/785270

**Comment by launchpad-issue-importer** [comment]
(by andresm-plast)

**Comment by launchpad-issue-importer** [comment]
(by jazzynico)
Confirmed on Ubuntu 10.10, Inkcape trunk revision 10220 (scour 0.26).
* Two of the three boxes' gradient are converted to flat color.
* The bottom left connector's marker is badly placed on the right of the left box.

**Comment by launchpad-issue-importer** [comment]
(by jazzynico)

**Comment by launchpad-issue-importer** [comment]
(by louis-simard-deactivatedaccount)
Rendered in Firefox 3.6.17/Linux, I do see that 2 of the 3 rounded rectangles have lost their white gradient, but I don't see a bad marker. Where is it?

**Comment by launchpad-issue-importer** [comment]
(by jazzynico)
The marker issue *may* be due to a connectors bug in Inkscape.
Attached, a screenshot taken on Windows XP with Inkscape r10221.

**Comment by Ede123** [comment]
Marker issue is definitely fixed (I made sure of that in 842123a39326223a91eb8e99ea8059e8158bc21c). 

The gradient is still removed. I reduced the testcase and tracked down the issue:
```SVG
<?xml version="1.0" encoding="UTF-8"?>
<svg height="600" width="600" xmlns="http://www.w3.org/2000/svg">
    <defs>
        <radialGradient id="a" cx="10%" cy="10%" r="100%" fx="10%" fy="10%">
            <stop offset="0%" stop-color="#ffffff" />
            <stop offset="100%" stop-color="#ffffcc" />
        </radialGradient>
        <radialGradient id="b" cx="10%" cy="10%" r="100%" fx="10%" fy="10%">
            <stop offset="0%" stop-color="#ffffff" />
            <stop offset="100%" stop-color="#ffffcc" />
        </radialGradient>
    </defs>
    <rect x="0" y="0" width="100" height="80" rx="10" ry="10" fill="url(#a) #ffffcc" />
    <rect x="0" y="0" width="100" height="80" rx="10" ry="10" fill="url(#b) #ffffcc" />
</svg>
```
The problem is the specification of a fallback background color in `fill="url(#b) #ffffcc"`.  
When scoured the identical gradients `#a` and `#b` are merged, but the reference is not updated properly. (As I was sceptical myself: This type of specification seems to be valid, see [SpecifyingPaint](https://www.w3.org/TR/SVG/painting.html#SpecifyingPaint), specifically the entry `<funciri> [ none | currentColor | <color> [<icccolor>] ]`
